docs/auth.md

```diff
@@ -0,0 +1,66 @@
+
+Just like the IoT provider case, the Producer SDK defines a few constructor APIs for creating file based credentials provider: https://github.com/awslabs/amazon-kinesis-video-streams-producer-c/blob/master/src/include/com/amazonaws/kinesis/video/common/Include.h#L602
+
+* createFileCredentialProvider - File based credential provider that reads the credentials from a specified file which gets periodically updated by an authentication module.
+* createFileCredentialProviderWithTime - Same API which also takes a time functionality.
+
+Similar to IoT credential provider, the file based credentials provider implements a caching policy in order to not read the credential file with each call.
+
+File based credential provider object is freed calling a single freeFileCredentialProvider API.
+
+#### Credentials Caching policy
+
+Credential providers like IoT perform "non-prompt" operation to fetch/refresh the credentials (i.e. the IoT credential provider will have to make a RESTful call to the IoT endpoint to retrieve those which is done in a synchronous manner). The low-level SDK core fetches the credentials multiple times and the credential provider will cache the retrieved credentials and submit those instead. 
+
+The providers will force-refresh the credentials if there is less than 38 seconds remaining from the cached credentials expiration. This constant is defined as: https://github.com/awslabs/amazon-kinesis-video-streams-producer-c/blob/master/src/source/Common/IotCredentialProvider.h#L25
+
+#### AWS public certificate integration
+
+AWS public certificate pem file is located at https://github.com/awslabs/amazon-kinesis-video-streams-producer-sdk-cpp/blob/master/certs/cert.pem
+
+Applications should include this pem file in their public certificate store in order for the networking client (libCurl or libWebSockets) to establish trust as the networking clients are configured to validate the server certificates. Each platform has their own default certificate paths - ex: on Linux based OS-es it's in /etc/ssl/, however, the applications could also choose to install the pem file in other locations and specify the path in the public APIs.
+
+C Producer: CA cert path can be specified in the CallbackProvider constructor APIs - ex: https://github.com/awslabs/amazon-kinesis-video-streams-producer-c/blob/master/src/include/com/amazonaws/kinesis/video/cproducer/Include.h#L297
+
+C++ Producer: By overriding getCertPath() virtual member function in the DeviceInfoProvider class https://github.com/awslabs/amazon-kinesis-video-streams-producer-sdk-cpp/blob/master/src/DeviceInfoProvider.h#L37
```

Comment:
In C++, we can also specify the path in CMake file this way: `CMakeLists.txt:53:add_definitions(-DKVS_CA_CERT_PATH="${CMAKE_CURRENT_SOURCE_DIR}/certs/cert.pem")`

Our current cmake file has this. It might be worth adding this method as well.

Reply:
Oh yeah